app/server/appsmith-server/src/main/java/com/appsmith/server/services/MockDataServiceImpl.java

```diff
@@ -119,8 +125,12 @@ private DatasourceConfiguration getMongoDataSourceConfiguration(String name, Moc
         List<Property> listProperty = new ArrayList<>();
         SSLDetails sslDetails = new SSLDetails();
 
-        MockDataCredentials credentials = mockDataSet.getCredentials().stream().filter(cred -> cred.getDbname().equals(name)).collect(Collectors.toList()).get(0);
+        List<MockDataCredentials> credentialsList = mockDataSet.getCredentials().stream().filter(cred -> cred.getDbname().equalsIgnoreCase(name)).collect(Collectors.toList());
+        if(credentialsList.size() == 0) {
+            return datasourceConfiguration;
```

Comment:
Instead of collecting into a list, I would suggest using `findFirst()`. An example is given below

```suggestion
        Optional<MockDataCredentials> credentials = mockDataSet.getCredentials().stream().filter(cred -> cred.getDbname().equalsIgnoreCase(name)).findFirst();
        // Check if credentials is available & not null
```